### PR TITLE
ProjectDisplayAndContactDev

### PR DESCRIPTION
**Pull request by DIEGOTEC18**
Modify SearchServlet and SearchedProject class to get and store all the project properties. Add JS functionality that caches the last searched projects. Add display project modal that displays the full project info. Add contact button that redirects the user to the project's owner profile page when clicked. Add styling to the displayed project modal. Minor styling changes.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 1]
```diff
@@ -60,19 +60,20 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
             //Get the properties of the entity:
             String title = (String) entity.getProperty("title");
-            String author = (String) entity.getProperty("author");
-            String collab = (String) entity.getProperty("collab");
-            String commitment = (String) entity.getProperty("commitment");
-            String difficulty = (String) entity.getProperty("difficulty");
+            String email = (String) entity.getProperty("owner_email");
+            String collab = (String) entity.getProperty("collabtyp");
```
Comment (Manniec) [inline src/main/java/com/google/sps/servlets/SearchServlet.java]: I just added the "e" collabtype per comments on my pull request

**Review by Manniec — COMMENTED** [review]
Cool! just need to resolve merge conflicts? (Sorry those were my fault, I changed the link to new project a bit in my last push)

**Comment by DIEGOTEC18** [comment]
@Manniec why did you merged master into my branch and not the other way around?

File: src/main/java/com/google/sps/servlets/SearchServlet.java (hunk 1 above)
Comment (DIEGOTEC18) [reply src/main/java/com/google/sps/servlets/SearchServlet.java]: Cool. I reflected on the changes in my code.

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 2]
```diff
@@ -97,19 +101,20 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
         //Create test Project Entity with dummy values:
         Entity projectEntity = new Entity("Project");
-        projectEntity.setProperty("title", "The principles of quantum computation.");
-        projectEntity.setProperty("description", "This is a test description");    
+        projectEntity.setProperty("title", "Quantic computation in a potato");
+        projectEntity.setProperty("description", "This is a test description but its about potatos.");
```
Comment (marcelpuyat) [inline src/main/java/com/google/sps/servlets/SearchServlet.java:105]: 🥔

File: src/main/webapp/search.js [hunk 3]
```diff
@@ -136,6 +143,59 @@ function buildProjectCard(title, tags, author, image){
 
 }
 
+function getSingleProject(projectId){
```
Comment (marcelpuyat) [inline src/main/webapp/search.js:146]: Well done and very well documented with the comments!

**Review by marcelpuyat — APPROVED** [review]
LGTM!

File: src/main/webapp/search.js (hunk 3 above)
Comment (DIEGOTEC18) [reply src/main/webapp/search.js:146]: Thanks!

**Pull request merged**